---
title: "**Sequencing 16S rRNA gene fragments using the PacBio SMRT DNA sequencing system**"
bibliography: references.bib
output:
  pdf_document:
    includes:
      in_header: header.tex
csl: peerj.csl
fontsize: 11pt
geometry: margin=1.0in
---


```{r knitr_settings, eval=TRUE, echo=FALSE, cache=FALSE}
opts_chunk$set("tidy" = TRUE)
opts_chunk$set("echo" = FALSE)
opts_chunk$set("eval" = TRUE)
opts_chunk$set("warning" = FALSE)
opts_chunk$set("cache" = FALSE)

inline_hook <- function(x){
	print(x)
	if(is.numeric(x)){
		if(abs(x - round(x)) < .Machine$double.eps^0.5){
			paste(format(x,big.mark=',', digits=0, scientific=FALSE))
		} else {
			paste(format(x,big.mark=',', digits=1, nsmall=1, scientific=FALSE))
		}

	} else {
    	paste(x)      
	}
}
knitr::knit_hooks$set(inline=inline_hook)

regions <- c("V1V3", "V1V5", "V1V6", "V1V9", "V3V5", "V4")
pretty_region <- c("V1V3"="V1-V3", "V1V5"="V1-V5", "V1V6"="V1-V6", "V1V9"="V1-V9", "V3V5"="V3-V5", "V4"="V4")

clrs <- rainbow(length(regions))
names(clrs) <- regions

library(rjson)
```

\vspace{35mm}

**Running title:** 16S rRNA genes sequencing with PacBio

\vspace{15mm}

**Authors:** Patrick D. Schloss^1#^, Sarah L. Westcott^1^, Matthew L. Jenior^1^,
and Sarah K. Highlander^2^

\vspace{40mm}

$\dagger$ To whom correspondence should be addressed: pschloss@umich.edu

1\. Department of Microbiology and Immunology, 1500 W. Medical Center, University of Michigan, Ann Arbor, MI 48109

2\. Genomic Medicine, J. Craig Venter Institute, 4120 Capricorn Lane, La Jolla, CA 92307


\newpage
\linenumbers


## Abstract [ Revise ]
Over the past 10 years, microbial ecologists have largely abandoned sequencing
16S rRNA genes by the Sanger sequencing method and have instead adopted highly
parallelized sequencing platforms. These new platforms, such as 454 and
Illumina's MiSeq, have allowed researchers to obtain millions of high quality,
but short sequences. These platforms have allowed researchers to significantly
improve the design of their experiments. The tradeoff has been the decline in
the number of full-length reference sequences that are deposited into databases.
To overcome this problem, we tested the ability of the PacBio Single Molecule,
Real-Time (SMRT) DNA sequencing platform to generate sequence reads from the 16S
rRNA gene. We generated sequencing data from the V4, V3-V5, V1-V3, V1-V6, and
V1-V9 variable regions from within the 16S rRNA gene from a synthetic mock
community and natural samples collected from human feces, mouse feces, and soil.
The synthetic mock community allowed us to assess the actual sequencing error
rate and how that error rate changed when different curation methods were
applied. We developed a simple method based on sequence characteristics and
quality scores to reduce the observed error rate for the V1-V9 region from 2.16%
to 0.32%. Unfortunately, this error rate was still 16-times higher than the
error rate that has been observed for the shorter reads generated by 454 and
Illumina's MiSeq sequencing platforms. Although the longer reads frequently
provided better classification, the wider adoption of this approach for 16S rRNA
gene sequencing is likely limited by its high sequencing error and low yield of
sequencing data relative to the other available platforms.


***Keywords:*** Microbial ecology, bioinformatics, sequencing error



## Introduction [ Revise ]
Advances in sequencing technologies over the past 10 years have introduced
considerable advances to the field of microbial ecology. Clone-based Sanger
sequencing of the 16S rRNA gene has largely been replaced by various platforms
produced by 454/Roche {e.g. Sogin, 2006 #1510}, Illumina
{e.g. Gloor, 2010 #2611}, and IonTorrent {e.g. Junemann, 2012 #2630}. It was
once common to sequence fewer than 100 16S rRNA gene sequences from several
samples using the Sanger approach {e.g. McCaig, 1999 #592}. Now it is common to
generate thousands of sequences from each of several hundred samples
{The Human Microbiome Consortium, 2012 #2617}. The advance in throughput has
come at the cost of read length. Sanger sequencing regularly generated 800 nt
per read and because the DNA was cloned, it was possible to obtain multiple
reads per fragment to yield a full-length sequence from a representative single
molecule. At approximately $8 (US) per sequencing read, most
researchers have effectively decided that full-length sequences are not worth
the increased cost relative to the cost of more recently developed approaches.
There is still a clear need to generate high-throughput full-length sequence
reads that are of sufficient quality that they can be used as references for
analyses based on obtaining short sequence reads.

Historically, all sequencing platforms were created to primarily perform genome
sequencing. When sequencing a genome, it is assumed that the same base of DNA
will be sequenced multiple times and the consensus of multiple sequence reads is
used to generate contigs. Thus, although an individual base call may have a high
error rate, the consensus sequence will have a low error rate. To sequence the
16S rRNA gene researchers use conserved primers to amplify a sub-region from
within the gene that is isolated from many organisms. Because the fragments
are not cloned, it is not possible to obtain high sequence coverage from the
same DNA molecule using these platforms. Thus, to reduce sequencing error rates
it has become imperative to develop stringent sequence curation and denoising
algorithms {Kozich, 2013 #2719; Schloss, 2011 #2466}. There has been a tradeoff
between read length, number of reads per sample, and the error rate. For
instance, we recently demonstrated that using the Illumina MiSeq and the 454
Titanium platforms the raw error rate varies between 1 and 2% {Kozich, 2013
#2719; Schloss, 2011 #2466}. Yet, it was possible to obtain error rates below
0.02% by adopting various denoising algorithms. However, the resulting fragments
were only 250-nt long. In the case of 454 Titanium, extending the length of the
fragment introduces length-based errors and in the case of the Illumina MiSeq,
increasing the length of the fragment reduces the overlap between the read pairs
reducing the ability of each read to mutually reduce the sequencing error.
Inadequate denoising of sequencing reads can have many negative effects
including limited ability to identify chimeras {Haas, 2011 #2404;Edgar, 2011 #2406}
and inflation of alpha- and beta-diversity metrics
{Huse, 2010 #2034; Kunin, 2010 #2259; Kozich, 2013 #2719; Schloss, 2011 #2466}.
Although MiSeq and 454 enjoy widespread use in the field, the MiSeq
platform is emerging as the leader because of the ability to sequence 15-20
million fragments that can be distributed across hundreds of samples for less
than $5000 (US).

As these sequencing platforms have grown in popularity, there has been a decline
in the number of full-length 16S rRNA genes being deposited into GenBank that
could serve as references. This is particularly frustrating since the
technologies have significantly improved our ability to detect and identify
novel populations for which we lack full-length reference sequences. A related
problem is the perceived limitation that the short reads generated by the 454
and Illumina platforms cannot be reliably classified to the genus or
species level. Previous investigators have utilized simulations to
demonstrate that increased read lengths usually increase the accuracy and
sensitivity of classification against reference databases
{Liu, 2008 #2188;Werner, 2012 #2474;Wang, 2007 #1809}. There is
clearly a need to develop sequencing technologies that will allow researchers to
generate high quality full-length 16S rRNA gene sequences in a high throughput
manner.

New advances in single molecule sequencing technologies, such as the platform
produced by Pacific Biosciences (PacBio), offer the opportunity to once again
obtain full-length sequence reads with a high depth of coverage from a large
number of samples. To this point, the PacBio Single Molecule, Real-Time (SMRT)
DNA Sequencing System has received limited application in the microbial ecology
research domain {Mosher, 2014 #3376; Fichot, 2013 #3453; Mosher, 2013 #3454}.
The SMRT system ligates hairpin adapters (i.e. SMRTbells) to the ends of
double-stranded DNA. Although the DNA molecule is linear, it is effectively
circularized allowing the sequencing polymerase to process around the molecule
multiple times {Au, 2012 #3455}. According to Pacific Biosciences the platform
is able to generate median read lengths longer than 8 kb with the P4-C2
chemistry; however, the single pass error rate is approximately 15%. Given the
circular nature of the DNA fragment, the full read length can be used to cover
the DNA fragment multiple times resulting in a reduced error rate. Therefore,
one should be able to obtain multiple coverage of the full 16S rRNA gene at a
reduced error rate.

Despite the opportunity to potentially generate high-quality full-length
sequences, the Pacific Biosciences platform has not been widely adopted for
sequencing 16S rRNA genes {Fichot, 2013 #3453; Mosher, 2014 #3376; Mosher, 2014 #3376}.
Previous studies utilizing the technology have removed reads with mismatched
primers and barcodes, ambiguous base  calls, and low quality scores {Fichot, 2013 #3453}.
Others have utilized the platform without describing the bioinformatic pipeline
that was utilized {Mosher, 2014 #3376; Mosher, 2014 #3376}. Regardless of the
curation methods, the error rates associated with sequencing the 16S rRNA gene
on the platform have never been reported. In the current study, we assessed the
quality of data generated by the Pacific Biosciences sequencer and whether it
could fill the need for generating high-quality, full-length sequence data. We
hypothesized that by modulating the 16S rRNA gene fragment length we could alter
the read depth and obtain reads longer than are currently available by the 454
and Illumina platforms but with the same quality. To test this hypothesis, we
developed a sequence curation pipeline that was optimized by reducing the
sequencing error rate of a mock bacterial community with known composition. The
resulting pipeline was then applied to 16S rRNA gene fragments that were
isolated from soil and human and mouse feces.


## Materials and Methods
***Community DNA.*** We utilized genomic DNA isolated from four communities.
These same DNA extracts were previously used to develop an Illumina MiSeq-based
sequencing strategy {Kozich, 2013 #2719}.  Briefly, we used a “Mock Community”
composed of genomic DNA from 21 bacterial strains: *Acinetobacter baumannii*
ATCC 17978, *Actinomyces odontolyticus* ATCC 17982, *Bacillus cereus* ATCC
10987, *Bacteroides vulgatus* ATCC 8482, *Clostridium beijerinckii* ATCC 51743,
*Deinococcus radiodurans* ATCC 13939, *Enterococcus faecalis* ATCC 47077,
*Escherichia coli* ATCC 70096, *Helicobacter pylori* ATCC 700392, *Lactobacillus
gasseri* ATCC 33323, *Listeria monocytogenes* ATCC BAA-679, *Neisseria
meningitidis* ATCC BAA-335, *Porphyromonas gingivalis* ATCC 33277,
*Propionibacterium acnes* DSM 16379, *Pseudomonas aeruginosa* ATCC 47085,
*Rhodobacter sphaeroides* ATCC 17023, *Staphylococcus aureus* ATCC BAA-1718,
*Staphylococcus epidermidis* ATCC 12228, *Streptococcus agalactiae* ATCC
BAA-611, *Streptococcus mutans* ATCC 700610, *Streptococcus pneumoniae* ATCC
BAA-334. The mock community DNA is available through BEI resources (v3.1,
HM-278D). Genomic DNAs from the three other communities were obtained using the
MO BIO PowerSoil DNA extraction kit. The human and mouse fecal samples were
obtained using protocols that were reviewed and approved by the University
Committee on Use and Care of Animals (Protocol #PRO00004877) and the
Institutional Review Board at the University of Michigan (Protocol
#HUM00057066). The human stool donor provided informed consent.

***Library generation and sequencing.*** [ Revise ]
The DNAs were each amplified in triplicate using barcoded primers targeting the
V4, V1-V3, V3-V5, V1-V5, V1-V6, and V1-V9 variable regions (Table 1). The
primers were synthesized so that the 5’ end of the forward and reverse primers
were each tagged with a 5-nt barcode sequence to allow multiplexing of samples
within a single sequencing run. Methods describing PCR, amplicon cleanup, and
pooling were described previously {Kozich, 2013 #2719}. The SMRTbell adapters
were ligated onto  the PCR products and the libraries were sequenced by Pacific
Biosciences using the P6-C24 chemistry on a PacBio RS II SMRT DNA Sequencing
System.


***Data analysis.***  [ Revise ]
All sequencing data were curated using mothur {Schloss, 2009 #1816} and analyzed
using the R programming language {R Core Team, 2014 #3456}. The raw data can be
obtained from the Sequence Read Archive at NCBI under accession SRP051686, which
are associated with BioProject PRJNA271568. Several specific features were
incorporated into mothur to facilitate the analysis of PacBio  sequence data.
First, because non-ambiguous base calls are assigned to Phred quality scores of
zero, the consensus fastq files were parsed so that scores of zero were
interpreted as corresponding to an ambiguous base call (i.e. N) in the
fastq.info command using the pacbio=T option. Second, because the consensus
sequence can be generated in the forward and reverse complement orientations, a
checkorient option was added to the trim.seqs command in order to identify the
proper orientation. These features were incorporated into mothur v.1.30. Because
chimeric molecules can be generated during PCR and would artificially inflate
the sequencing error, it was necessary to remove these data prior to assessing
the error rate. Because we knew the true sequences for the strains in the mock
community we could calculate all possible chimeras between strains in the mock
community (*in silico* chimeras). If a sequence read was 3 or more nucleotides
more similar to an *in silico* chimera than it was to a non-chimeric reference
sequence, it was classified as a chimera and removed from further consideration.
Identification of *in silico* chimeras and calculation of sequencing error rates
was performed using the seq.error command in mothur {Schloss, 2011 #2466}. *De
novo* chimera detection was also performed on the mock and other sequence data
using the abundance-based algorithm implemented in UCHIME {Edgar, 2011 #2406}.
Sequences sequences were aligned against a SILVA-based reference alignment
{Pruesse, 2007 #1735} using a profile-based aligner {Schloss, 2009 #1825} and
were classified against the SILVA {Pruesse, 2007 #1735}, RDP {Cole, 2014 #3460},
and greengenes {Werner, 2012 #2474} reference taxonomies  using a naive Bayesian
classifier {Wang, 2007 #1809}. Sequences were assigned to operational taxonomic
units using the average neighbor clustering algorithm with a 3% distance
threshold {Schloss, 2011 #3156}.  Detailed methods including this paper as an R
markdown file are available as a public online repository
(http://github.com/SchlossLab/ Schloss_PacBio16S_PeerJ_2015).


## Results and Discussion

```{r error_profile}
error_profile <- fromJSON(scan("data/process/error_profile.json", quiet=T, what="character"))
```

***The PacBio error profile and a basic sequence curation procedure.***
To build a sequence curation pipeline, we first needed to characterize the error rate associated with sequencing the 16S rRNA gene. We observed an average sequencing error rate of `r format(error_profile$overall_error, digits=2, nsmall=2)`%. Insertions, deletions, and substitutions accounted for `r format(error_profile$ins_rate, digits=3, nsmall=1)`, `r format(error_profile$del_rate, digits=3, nsmall=1)`, and `r format(error_profile$subs_rate, digits=3, nsmall=1)`% of the errors, respectively. The substitution errors were equally likely and all four bases were equally likely to cause insertion errors. Interestingly, guanines (`r format(error_profile$del_bias$dG, digits=3, nsmall=1)`%) and adenines (`r format(error_profile$del_bias$dA, digits=3, nsmall=1)`%) were more likely to be deleted than cytosines (`r error_profile$del_bias$dC, digits=3, nsmall=1)`%) or thymidines (`r error_profile$del_bias$dT, digits=2, nsmall=1)`%). The PacBio quality values varied between `r paste(error_profile$qual_range, collapse=' and ')`. Surprisingly, the percentage of base calls that had the maximum quality value did not vary among correct base calls (`r format(error_profile$percent_over_93$m, digits=3, nsmall=1)`%), substitutions (`r format(error_profile$percent_over_93$s, digits=3, nsmall=1)`%), and insertions (`r format(error_profile$percent_over_93$i, digits=3, nsmall=1)`%). Although the individual base quality scores could not be used to screen sequence quality, we observed a strong correlation between our observed error rate and the predicted error rate as calculated by the PacBio software (Pearson's R: `r format(error_profile$pred_error_corr, digits=2, nsmall=2)`; Figure 1A).

```{r basic}
error_summary <- read.table(file="data/process/error_summary.tsv", header=T)

raw_to_basic_lost <- 100 * (1 - error_summary[error_summary$method == "basic","nseqs"]/error_summary[error_summary$method == "raw","nseqs"])
```

We established a simple curation procedure by culling any sequence that had a string of the same base repeated more than 8 times  or did not start and end at the expected alignment coordinates for that region of the 16S rRNA gene. This reduced the experiment-wide error rate from `r format(mean(error_summary[error_summary$method=="raw","error"]), digits=3, nsmall=2)` to `r format(mean(error_summary[error_summary$method=="basic","error"]), digits=2, nsmall=2)`%. This basic procedure resulted in the removal of between `r format(min(raw_to_basic_lost), digits=3, nsmall=1)` (`r pretty_region[names(which.min(raw_to_basic_lost))]`) and `r format(max(raw_to_basic_lost), digits=3, nsmall=1)` (`r pretty_region[names(which.max(raw_to_basic_lost))]`)% of the reads (Table 2). Although the percentage of reads removed increased with the length of the fragment, there was no obvious relationship between fragment length and error rate (Figure 2).


***Identifying correlates of increased sequencing error.***
In contrast to the 454 and Illumina-based platforms where the sequencing quality decays with length, the consensus sequencing approach employed by the PacBio sequencer is thought to generate a uniform distribution of errors. This makes it impossible to simply trim sequences to high quality regions. Therefore, we sought to identify characteristics within sequences that would allow us to identify and remove those sequences with errors using three different approaches. First, we hypothesized that errors in the barcode and primer would be correlated with the error rate for the entire sequence. We observed a strong relationship between the number of mismatches to the barcodes and primers and the error rate of the rest of the sequence fragment (Figure 1B). Although allowing no mismatches to the barcodes and primers yielded the lowest error rate, that stringent criterion removed a large fraction of the reads from the dataset. Allowing at most one mismatch only marginally increased the error rate while retaining more sequences in the dataset (Figure 2). Second, we hypothesized that increased sequencing coverage should yield lower error rates. We found that once we had obtained 10-fold coverage of the fragments, the error rate did not change appreciably (Figure 1C). When we compared the error rates of reads with at least 10-fold coverage to those with less coverage, we reduced the error rate by `r paste(format(range(error_summary[error_summary$method == "coverage","error_red"]), digits=3, nsmall=1), collapse=" to ")`% (Figure 2). Third, based on the observed correlation between the predicted and observed error rates, we sought to identify a minimum predicted error rate that would allow us to reduce the observed error rate. The average observed error rate for sequences with predicted error rates between 0.01 and 0.10% was linear (Figure 1D). We decided to use a threshold of 0.01% because a large number of sequence reads were lost when we used a smaller threshold. When we used this threshold, we were able to reduce the error rate by `r paste(format(range(error_summary[error_summary$method == "pred","error_red"]), digits=3, nsmall=1), collapse=" to ")`% (Figure 2A). Finally, we quantified the effect of combining filters. We found that any combination of filters that included the predicted error rate threshold had the most significant impact on reducing the observed error rate. Furthermore, the inclusion of the mismatch and coverage filters had a negligible impact on error rates, but had a significant impact on the number of sequences included in the analysis. For instance, among the V1-V9 data, requiring sequences to have a predicted error rate less than 0.01% resulted in a `r format(error_summary[error_summary$method == "pred" & error_summary$region == "V1V9","error_red"], digits=3, nsmall=1)`% reduction in error and resulted in the removal of `r format(error_summary[error_summary$method == "pred" & error_summary$region == "V1V9","nseqs_red"], digits=3, nsmall=1)`% of the sequences. Adding the mismatch or coverage filter had no effect on the reduction of error, but resulted in the removal of `r format(error_summary[error_summary$method == "pb_pred" & error_summary$region == "V1V9","nseqs_red"], digits=3, nsmall=1)` and `r format(error_summary[error_summary$method == "coverage_pred" & error_summary$region == "V1V9","nseqs_red"], digits=3, nsmall=1)` % of the sequences, respectively. Use of all filters had no impact on the reduction in the observed error rate, but resulted in the removal of `r format(error_summary[error_summary$method == "all" & error_summary$region == "V1V9","nseqs_red"], digits=3, nsmall=1)`% of the sequences. The remainder of this paper only uses sequences with a predicted error rate less than 0.01%.


***Pre-clustering sequences to further reduce sequencing noise.***
Previously, we implemented a pre-clustering algorithm where sequences were sorted by their abundance in decreasing order and rare sequences are clustered with a more abundant sequence if the rare sequences have fewer mismatches than a defined threshold when compared to the more abundant sequence [@refs]. The recommended threshold was a 1-nt difference per 100-nt of sequence data. For example, the threshold for 250 bp fragment from the V4 region would be 2 nt or 14 for the 1458 bp V1-V9 fragments. This approach removes residual PCR and sequencing errors while not overwhelming the resolution needed to identify OTUs that are based on a 3% distance threshold. The tradeoff of this approach is that one would unable to differentiate V1-V9 sequences that truly differed by less than 14 nt. When we applied this approach to our PacBio data, we observed a reduction in the error rate between `r format(min(error_summary[error_summary$method == "precluster","nseqs_red"]), digits=3, nsmall=1)` (`r pretty_region[which.min(error_summary[error_summary$method == "precluster","nseqs_red"])]`) and `r format(max(error_summary[error_summary$method == "precluster","nseqs_red"]), digits=3, nsmall=1)`% (`r pretty_region[which.max(error_summary[error_summary$method == "precluster","nseqs_red"])]`). The final error rates varied between `r round(min(error_summary[error_summary$method == "precluster","error"]), digits=2)` (`r pretty_region[which.min(error_summary[error_summary$method == "precluster","error"])]`) and `r round(max(error_summary[error_summary$method == "precluster","error"]), digits=2)`% (`r pretty_region[which.max(error_summary[error_summary$method == "precluster","error"])]`). The full-length (i.e. V1-V9) fragments had an error rate of `r round(error_summary[error_summary$method == "precluster" & error_summary$region=="V1V9","error"], digits=2)`% (Figure 2B), this is similar to what we have previously observed using the 454 and Illumina MiSeq platforms (0.02%){Schloss, 2011 #2466; Kozich, 2013 #2719}


```{r}
sobs_data <- read.table(file="data/process/sobs_table.tsv", header=T)
```

***Effects of error rates on OTU assignments.***
The sequencing error rate is known to affect the number of OTUs that are observed {Schloss, 2011 #2466}. For each region, we determined that if there were no chimeras or PCR or sequencing errors, then we would expect to find `r perfect.otu[[1]]` OTUs. When achieved perfect chimera removal, but allowed for PCR and sequencing errors, we observed between `r diff<-otu.table[,2]-perfect.otu; round(min(diff), digits=1)` (`r names(which.min(diff))`) and `r round(max(diff), digits=1)` (`r names(which.max(diff))`) extra OTUs. The range in the number of extra OTUs was largely explained by the sequencing error rate (Pearson's R=`r round(cor.test(diff, pooled_error[,"precluster"])$estimate, 2)`). Next, we determined the number of OTUs that were observed when we used UCHIME to identify chimeric sequence. Under these more realistic conditions, we observed between `r diff<-otu.table[,3]-perfect.otu; round(min(diff), digits=1)` (`r names(which.min(diff))`) and `r round(max(diff), digits=1)` (`r names(which.max(diff))`) extra OTUs. Finally, we calculated the number of OTUs in the soil, mouse, and human samples using the same pipeline with chimera detection and removal based on the UCHIME algorithm. Surpisingly, there was not a clear relationship across sample type and region. Again, we found that there was a strong correlation between the number of observed OTUs and the error rate for the mouse (R=`r format(round(cor.test(otu.table[,"Mouse"], pooled_error[,"precluster"])$estimate, 2), nsmall=2)`) and human samples (R=`r format(round(cor.test(otu.table[,"Human"], pooled_error[,"precluster"])$estimate, 2), nsmall=2)`). These results underscore the effect of sequencing error on the inflation of the number of observed OTUs.


```{r, classificationSummary}
taxonomy_data <- read.table(file="data/process/taxonomy_depth_analysis.tsv", header=T)
```

***Increasing sequence length improves classification.***
We classified all of the sequence data we generated using the naïve Bayesian classifier using the RDP, SILVA, and greengenes reference taxonomies (Figure 3). In general, increasing the length of the region improved the ability to assign the sequence to a genus or species. Interestingly, each of the samples we analyzed varied in the ability to assign its sequences to the depth of genus or species. Furthermore, the reference database that did the best job of classifying the sequences varied by sample type. For example, the SILVA reference did the best for the human feces and soil samples and the RDP did the best for the mouse feces samples. An advantage of the greengenes database is that it contains information for 2,514 species-level lineages for 11% of the reference sequences; the other databases only provided taxonomic data to the genus level. There was a modest association between the length of the fragment and the ability to classify sequences to the species-level for the human samples; there was no such association for the mouse and soil samples. In fact, at most `r format(round(100*max(taxonomy_data[taxonomy_data$db=="gg" & taxonomy_data$dataset=="soil","X7"]), digits=1), nsmall=1)`% of the soil sequences and `r format(round(100*max(taxonomy_data[taxonomy_data$db=="gg" & taxonomy_data$dataset=="mouse","X7"]), digits=1), nsmall=1)`% of the mouse sequences could be classified to a species. These results indicate that the ability to classify sequences to the genus or species level is a function of read length, sample type, and the reference database.  


```{r}
one_offs <- read.table(file="data/process/non_random_analysis.tsv", header=T)
percent_one_off <- 100 * prop.table(as.matrix(one_offs), 2)

get_max_one_off <- function(region){
	max(as.numeric(rownames(one_offs[one_offs[,region]==1,])))
}
max_one_off <- sapply(regions, get_max_one_off)
```

***Sequencing errors are not random.***
Above, we described that although there was no obvious bias in the substitution or insertion rate, we did observe that guanines and adenines were more likely to be deleted than cytosines or thymidines. This lack of randomness in the error profile suggested that there might be a systematic non-random distribution of the errors across the sequences. This would manifest itself by the creation of duplicate sequences with the same error. We identified all of the mock community sequences that had a 1-nt difference to the true sequence. For these three regions, between `r paste0(format(round(range(percent_one_off[1,]), digits=1), nsmall=1), collapse=" and ")` of the sequences with 1-nt errors were only observed once. We found that the frequency of the most abundant 1-nt error paralleled the number of sequences. Surprisingly, the same 1-nt error appeared `r max_one_off["V1V6"]` times (`r format(round(percent_one_off[max_one_off["V1V6"], "V1V6"], digits=2), nsmall=2)`%) in the V1-V6 mock data and another 1-nt error appeared `r max_one_off["V1V9"]` times (`r format(round(percent_one_off[max_one_off["V1V9"], "V1V9"], digits=2), nsmall=2)`%) in the V1-V9 mock data. Contrary to previous reports {Carneiro, 2012 #3458;Koren, 2012 #3459}, these results indicate that reproducible errors occur with the PacBio sequencing platform and that they can be quite abundant. Through the use of the pre-clustering step described above these 1-nt errors would be ameliorated; however, this result indicates that caution should be used when attempting to use fine-scale OTU definitions.


## Conclusions
The various sequencing platforms that are available to microbial ecologists are able to fill unique needs and have their own strengths and weaknesses. For sequencing the 16S rRNA gene, the 454 platform is able to generate a moderate number of high-quality 500-nt sequence fragments (error rates below 0.02%) {Schloss, 2011 #2466} and the MiSeq platform is able to generate a large number of high-quality 250-nt sequence fragments (error rates below 0.02%) {Kozich, 2013 #2719}. The promise of the PacBio sequencing platform was the generation of high-quality near full-length sequence fragments. As we have shown in this study, it is possible to generate near full-length sequences with error rates that are slightly higher, but comparable to the other platforms (i.e. `r round(error_summary[error_summary$region=="V1V9" & error_summary$method=="precluster","error"],2)`%). With the exception of the V4 region (`r round(error_summary[error_summary$region=="V4" & error_summary$method=="precluster","error"], 2)`%), the error rates were less than `r round(max(error_summary[error_summary$region!="V4" & error_summary$method=="precluster","error"]), 2)`%. When we considered the shorter V4 region, which is similar in length to what is sequenced by the MiSeq platform, the error rates we observed with the PacBio platform were nearly `r round(error_summary[error_summary$region=="V4" & error_summary$method=="precluster","error"]/0.02)`-fold higher than what has previously been reported on the other platforms. It was unclear why these shorter reads had such a high error rate relative to the other regions. At this point, the primary limitation of generating full-length sequences on the PacBio platform is the cost of generating the data and accessibility to the sequencers.

```{r binomial_analysis}
unique <- error_summary[error_summary$region=="V1V9" & error_summary$method=="coverage_pred","error"]/100

precluster <- error_summary[error_summary$region=="V1V9" & error_summary$method=="precluster","error"] / 100
unique_to_precluster <- unique / precluster
```

The widespread adoption of the 454 and MiSeq platforms and decrease in the use of Sanger sequencing for the 16S rRNA gene has resulted in a decrease in the generation of the full-length reference sequences that are needed for performing phylogenetic analyses and designing lineage specific PCR primers and fluorescent *in situ* hybridization (FISH) probes. It remains to be determined whether the error rates we observed for full-length sequences are prohibitive for these applications. We can estimate the distribution of errors assuming that the errors follow a binomial distribution along the length of the 1,500 nt gene with the error rate that we achieved from the V1-V9 mock community data prior to pre-clustering the sequences, which was `r round(100*unique,2)`% (Figure 5). Under these conditions one would expect `r round(100*dbinom(0, 1500, unique), 2)`% of the sequences to have no errors and 50% of the sequences would have at least `r qbinom(0.50, 1500, unique)` errors. After applying the pre-clustering denoising step, the error rate drops by `r round(unique_to_precluster, 1)`-fold to `r 100*precluster`%. With this error rate, we would expect `r round(100*dbinom(0, 1500, precluster), 2)`% of the sequences to have no sequencing errors. The cost of the reduced error rate is the loss of resolution among closely related sequences.

Full-length sequences are frequently seen as a panacea to overcome the limitations of taxonomic classifications. The ability to classify each of our sample types benefited from the generation of full-length sequences. It was interesting that the benefit varied by sample type and database. For example, using the mouse libraries, the ability to classify each of the regions differed by less than 5% when classifying against the SILVA and greengenes databases. The effect of the database that was used was also interesting. The RDP database outperformed the other databases for the mouse samples and the SILVA database outperformed the others for the human and soil samples. The three databases were equally effective for classifying the mock community. Finally, since only the greengenes database provided species-level information for its reference sequences it was the only database that allowed for resolution of species-level classification. The sequences from the mouse and soil libraries were not effectively classified to the species level (all less than 10%). In contrast, classification of the human libraries resulted in more than 40% of the sequences being classified to a genus, regardless of the region. That the variation in species-level classification for the human libraries was less than 10% suggests that the benefit of added length is minimal considering the lower sequencing yield.

The development of newer sequencing technologies continue to advance and there is justifiable excitement to apply these technologies to sequence the 16S rRNA gene. Although it is clearly possible to generate sequencing data from these various platforms, it is critical that we assess the platforms for their ability to generate high quality data and the particular niche that the new approach will fill. With this in mind, it is essential that researchers utilize mock communities as part of their experimental design so that they can quantify their error rates. The ability to generate near full-length 16S rRNA gene sequences is an exciting advance that will hopefully expand our ability to improve the characterization of microbial communities.



## Acknowledgements
The Genomic DNA from Microbial Mock Community A (Even, Low Concentration, v3.1, HM-278D) was obtained through the NIH Biodefense and Emerging Infections Research Resources Repository, NIAID, NIH as part of the Human Microbiome Project.


## Funding statement
This study was supported by grants from the NIH (R01HG005975, R01GM099514 and P30DK034933 to PDS and U54HG004973 to SKH).


## Tables

**Table 1. Summary of the primer pairs used to generate the 16S rRNA gene fragment fragments and the characteristics of each region.**

```{r Table_1, warning=TRUE, results='asis'}

getSloppyRevComp <- function(sequence){
	rev.sequence <- paste(rev(unlist(strsplit(sequence, ""))), collapse="")
	revcomp.sequence <- chartr("ATGCRYSWKMBDHVN", "TACG...........", rev.sequence)
	return(revcomp.sequence)
}

getPos <- function(primer, sequence=ecoli, forward=TRUE){
	if(forward==FALSE){
		primer <- getSloppyRevComp(primer)
	}
	primer <- gsub("[^ATGC]", ".", primer)

	pos <- regexpr(primer, sequence)

	if(forward==FALSE){
		pos <- pos + nchar(primer) -1
	}
	return(pos)

}

#taken from... http://www.ncbi.nlm.nih.gov/nuccore/NC_022648.1?report=fasta&from=4564120&to=4565661
ecoli <- "AAATTGAAGAGTTTGATCATGGCTCAGATTGAACGCTGGCGGCAGGCCTAACACATGCAAGTCGAACGGTAACAGGAAGCAGCTTGCTGCTTTGCTGACGAGTGGCGGACGGGTGAGTAATGTCTGGGAAACTGCCTGATGGAGGGGGATAACTACTGGAAACGGTAGCTAATACCGCATAACGTCGCAAGACCAAAGAGGGGGACCTTCGGGCCTCTTGCCATCGGATGTGCCCAGATGGGATTAGCTAGTAGGTGGGGTAACGGCTCACCTAGGCGACGATCCCTAGCTGGTCTGAGAGGATGACCAGCCACACTGGAACTGAGACACGGTCCAGACTCCTACGGGAGGCAGCAGTGGGGAATATTGCACAATGGGCGCAAGCCTGATGCAGCCATGCCGCGTGTATGAAGAAGGCCTTCGGGTTGTAAAGTACTTTCAGCGGGGAGGAAGGGAGTAAAGTTAATACCTTTGCTCATTGACGTTACCCGCAGAAGAAGCACCGGCTAACTCCGTGCCAGCAGCCGCGGTAATACGGAGGGTGCAAGCGTTAATCGGAATTACTGGGCGTAAAGCGCACGCAGGCGGTTTGTTAAGTCAGATGTGAAATCCCCGGGCTCAACCTGGGAACTGCATCTGATACTGGCAAGCTTGAGTCTCGTAGAGGGGGGTAGAATTCCAGGTGTAGCGGTGAAATGCGTAGAGATCTGGAGGAATACCGGTGGCGAAGGCGGCCCCCTGGACGAAGACTGACGCTCAGGTGCGAAAGCGTGGGGAGCAAACAGGATTAGATACCCTGGTAGTCCACGCCGTAAACGATGTCGACTTGGAGGTTGTGCCCTTGAGGCGTGGCTTCCGGAGCTAACGCGTTAAGTCGACCGCCTGGGGAGTACGGCCGCAAGGTTAAAACTCAAATGAATTGACGGGGGCCCGCACAAGCGGTGGAGCATGTGGTTTAATTCGATGCAACGCGAAGAACCTTACCTGGTCTTGACATCCACGGAAGTTTTCAGAGATGAGAATGTGCCTTCGGGAACCGTGAGACAGGTGCTGCATGGCTGTCGTCAGCTCGTGTTGTGAAATGTTGGGTTAAGTCCCGCAACGAGCGCAACCCTTATCCTTTGTTGCCAGCGGTCCGGCCGGGAACTCAAAGGAGACTGCCAGTGATAAACTGGAGGAAGGTGGGGATGACGTCAAGTCATCATGGCCCTTACGACCAGGGCTACACACGTGCTACAATGGCGCATACAAAGAGAAGCGACCTCGCGAGAGCAAGCGGACCTCATAAAGTGCGTCGTAGTCCGGATTGGAGTCTGCAACTCGACTCCATGAAGTCGGAATCGCTAGTAATCGTGGATCAGAATGCCACGGTGAATACGTTCCCGGGCCTTGTACACACCGCCCGTCACACCATGGGAGTGGGTTGCAAAAGAAGTAGGTAGCTTAACCTTCGGGAGGGCGCTTACCACTTTGTGATTCATGACTGGGGTGAAGTCGTAACAAGGTAACCGTAGGGGAACCTGCGGTTGGATCACCTCCTTA"



oligos <- read.table(file="data/references/pacbio.oligos", stringsAsFactors=F)[1:6,]
rownames(oligos) <- pretty_region[oligos$V4]
oligos <- oligos[,c("V2", "V3")]
colnames(oligos) <- c("Forward", "Reverse")

start <- sapply(oligos[,1], getPos)
end <- sapply(oligos[,2], getPos, forward=FALSE)


coordinates <- paste(start, end, sep="-")
length <- end - start - nchar(oligos[,1]) - nchar(oligos[,2]) + 1
t1 <- cbind(oligos, coordinates, length)
knitr::kable(t1, digits=1, row.names=TRUE, col.names=c("Forward", "Reverse", "*E. coli* coordinates^a^", "Length (bp)^b^"), align=c("l", "l", "c", "c"))
```

^a^	The coordinates where the start and end of the forward and reverse primers anneal, respectively.

^b^	The number of bases between the primers.



**Table 2. Summary of the reasons that sequences were excluded because of the basic sequence curation steps**

```{r Table_2, warning=TRUE, results='asis'}
rownames(reasons.lost) <- pretty_region[rownames(reasons.lost)]
reasons.lost <- cbind(orig.nseqs, 100*reasons.lost, basic.nseqs)
knitr::kable(reasons.lost, digits=1, row.names=TRUE,
						 col.names=c("Initial sequences (N)", "Good reads (%)",
						 						"Wrong start/end position (%)",
						 						"Excessively long homopolymers (%)",
						 						"Ambiguous base calls (%)",
						 						"Sequences remaining (N\\)"),
						 align="c")
```


## Figures

**Figure 1. Summary of errors in data generated using PacBio sequencing platform
to sequence various regions within the 16S rRNA gene.** The predicted error rate using PacBio's sequence analysis algorithm correlated well with the observed error rate (Pearson's R: `r format(error_profile$pred_error_corr, digits=2, nsmall=2)`; A). Because of the large number of sequences, we randomly selected 5% of the data to show in panel A. The sequencing error rate of the amplified gene fragments
increased with mismatches to the barcodes and primers (B). The sequencing error
rate declined with increased sequencing coverage; however, increasing the
sequencing depth beyond 10-fold coverage had no meaningful effect on the
sequencing error rate (C).


**Figure 2. Change in error rate (A) and the percentage of sequences that were
retained (B) when using various sequence curation methods.** The condition that
was used for downstream analyses is indicated by the star. The plotted numbers
represent the region that was sequenced. For example "15" represents the data
for the V1-V5 region.


**Figure 3. Percentage of unique sequences that could be classified.**
Classifications were performed using taxonomy references curated from the RDP,
SILVA, or greengenes databases for the four types of samples that were sequenced
across the six regions from the 16S rRNA gene. Only the greengenes taxonomy
reference provided species-level information.






### Figure 3

```{r, Figure_4, fig.width=7, fig.height=7}
rdp <- 100 * cbind("Mock"=data[data$db=="pds" & data$dataset=="mock", "genus_species"],
				 "Human"=data[data$db=="pds" & data$dataset=="human", "genus_species"],
				 "Mouse"=data[data$db=="pds" & data$dataset=="mouse", "genus_species"],
				 "Soil"=data[data$db=="pds" & data$dataset=="soil", "genus_species"])
rownames(rdp) <- c("V4", "V3-V5", "V1-V3", "V1-V5", "V1-V6", "V1-V9")

gg <- 100 * cbind("Mock"=data[data$db=="gg" & data$dataset=="mock", "genus_species"],
				"Human"=data[data$db=="gg" & data$dataset=="human", "genus_species"],
				"Mouse"=data[data$db=="gg" & data$dataset=="mouse", "genus_species"],
				"Soil"=data[data$db=="gg" & data$dataset=="soil", "genus_species"])
rownames(gg) <- c("V4", "V3-V5", "V1-V3", "V1-V5", "V1-V6", "V1-V9")

silva <- 100 * cbind("Mock"=data[data$db=="nr" & data$dataset=="mock", "genus_species"],
					 "Human"=data[data$db=="nr" & data$dataset=="human", "genus_species"],
					 "Mouse"=data[data$db=="nr" & data$dataset=="mouse", "genus_species"],
					 "Soil"=data[data$db=="nr" & data$dataset=="soil", "genus_species"])
rownames(silva) <- c("V4", "V3-V5", "V1-V3", "V1-V5", "V1-V6", "V1-V9")

gg.sp <- 100 * cbind("Mock"=data[data$db=="gg" & data$dataset=="mock", "X7"],
					 "Human"=data[data$db=="gg" & data$dataset=="human", "X7"],
					 "Mouse"=data[data$db=="gg" & data$dataset=="mouse", "X7"],
					 "Soil"=data[data$db=="gg" & data$dataset=="soil", "X7"])
rownames(gg.sp) <- c("V4", "V3-V5", "V1-V3", "V1-V5", "V1-V6", "V1-V9")

par(mar=c(5, 5, 0.5, 1))
dotchart(gg, xlim=c(0,100), col="black", xlab="Unique reads that classified\nto genus or species level (%)", pch=19)

points(x=rdp[,"Mock"], y=25:30, pch=15, col="black")
points(x=rdp[,"Human"], y=17:22, pch=15, col="black")
points(x=rdp[,"Mouse"], y=9:14, pch=15, col="black")
points(x=rdp[,"Soil"], y=1:6, pch=15, col="black")

points(x=silva[,"Mock"], y=25:30, pch=17, col="black")
points(x=silva[,"Human"], y=17:22, pch=17, col="black")
points(x=silva[,"Mouse"], y=9:14, pch=17, col="black")
points(x=silva[,"Soil"], y=1:6, pch=17, col="black")

points(x=gg.sp[,"Mock"], y=25:30, pch=21, bg="gray")
points(x=gg.sp[,"Human"], y=17:22, pch=21, bg="gray")
points(x=gg.sp[,"Mouse"], y=9:14, pch=21, bg="gray")
points(x=gg.sp[,"Soil"], y=1:6, pch=21, bg="gray")

legend(x=63, y=12, legend=c("RDP (gen.)", "SILVA (gen.)", "greengenes (gen.+sp.)", "greengenes (sp.)"), pch=c(15, 17, 19, 21), col="black", pt.bg=c("black", "black", "black", "gray"), bg="white")
```


#plot one_offs
```{r}
one_offs <- read.table(file="data/process/non_random_analysis.tsv", header=T)

regions <- c("V1V3", "V1V5", "V1V6", "V1V9", "V3V5", "V4")

z <- prop.table(as.matrix(one_offs), 2)[1:10,]
plot(NA, xlim=c(1,10), ylim=c(0,1), xlab="", ylab="", axes=F)
for(r in regions){
	points(1:10, z[,r], type="l")
}
axis(1, at=1:10, label=1:10)
axis(2)
```


## References
